pkg/remote/crd_upgrade.go

```diff
@@ -21,6 +29,142 @@ func PatchCRD(ctx context.Context, clnt client.Client, crd *v1extensions.CustomR
 		client.FieldOwner(v1beta2.OperatorName))
 }
 
+type CrdType string
+
+const (
+	KCP CrdType = "KCP"
+	SKR CrdType = "SKR"
+)
+
+func updateRemoteCRD(
+	ctx context.Context, plural string, kyma *v1beta2.Kyma, runtimeClient Client, controlPlaneClient Client) (
+	bool, *v1extensions.CustomResourceDefinition, *v1extensions.CustomResourceDefinition, error) {
+	crd := &v1extensions.CustomResourceDefinition{}
+	crdFromRuntime := &v1extensions.CustomResourceDefinition{}
+	var err error
+	err = controlPlaneClient.Get(
```

Comment:
Since CRD in KCP can be only updated when we deploy LM, for performance purposes, we can cache the CRD generation, because during the runtime of LM, CRD wouldn't get updated, we can save many remote calls in this case. But this can be done in the future issue. https://github.com/kyma-project/lifecycle-manager/issues/598